src/Vex/Bindings.cpp

```diff
@@ -42,13 +45,44 @@ void ValidateBufferBinding(const BufferBinding& binding, BufferUsage::Flags vali
 
     if (usage == BufferBindingUsage::StructuredBuffer || usage == BufferBindingUsage::RWStructuredBuffer)
     {
-        if (!binding.strideByteSize.has_value())
-        {
-            VEX_LOG(Fatal,
-                    "Invalid binding for resource \"{}\": In order to use a binding as a structured buffer, you must "
-                    "pass in a valid stride.",
-                    buffer.desc.name);
-        }
+        VEX_CHECK(binding.strideByteSize.has_value(),
+                  "Invalid binding for resource \"{}\": In order to use a binding as a structured buffer, you must "
+                  "pass in a valid stride.",
+                  buffer.desc.name);
+
+        VEX_CHECK(*binding.strideByteSize > 0,
+                  "Invalid binding for resource \"{}\": Stride for structured buffers must not be 0.",
+                  buffer.desc.name);
+
+        u64 offsetByteSize = binding.offsetByteSize.value_or(0);
+        VEX_CHECK(offsetByteSize % *binding.strideByteSize == 0,
+                  "Invalid binding for resource \"{}\": Offset must be a multiple of the stride.",
+                  buffer.desc.name);
+
+        VEX_CHECK(
+            binding.rangeByteSize.value_or(binding.buffer.desc.byteSize - offsetByteSize) % *binding.strideByteSize ==
+                0,
+            "Invalid binding for resource \"{}\": Range must be a multiple of the stride.",
+            buffer.desc.name);
+    }
+
+    if (usage == BufferBindingUsage::ConstantBuffer)
+    {
+        VEX_CHECK(binding.offsetByteSize.value_or(0) % ConstantBufferBindingOffsetMultiple == 0,
+                  "Constant buffer offsets must be a multiple of 256 bytes")
+    }
+
+    if (usage == BufferBindingUsage::ByteAddressBuffer || usage == BufferBindingUsage::RWByteAddressBuffer)
+    {
+        VEX_CHECK(binding.offsetByteSize.value_or(0) % ByteAddressBufferOffsetMultiple == 0,
+                  "ByteAddressBuffer offsets must be a multiple of {} bytes (elements are {} bytes wide)",
+                  ByteAddressBufferOffsetMultiple,
+                  ByteAddressBufferOffsetMultiple)
+
+        VEX_CHECK(binding.rangeByteSize.value_or(0) % ByteAddressBufferOffsetMultiple == 0,
+                  "ByteAddressBuffer range must be a multiple of {} bytes (elements are {} bytes wide)",
+                  ByteAddressBufferOffsetMultiple,
+                  ByteAddressBufferOffsetMultiple)
```

Comment:
Please keep the "Invalid binding for resource {}" error message formatting for this validation code.